addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/handler/YeelightHandlerBase.java

```diff
@@ -0,0 +1,231 @@
+import static org.openhab.binding.yeelight.YeelightBindingConstants.COLOR_TEMPERATURE_STEP;
+
+/**
+ * The {@link YeelightHandlerBase} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Coaster Li (lixin@yeelink.net) - Initial contribution
+ */
+public abstract class YeelightHandlerBase extends BaseThingHandler
+        implements DeviceConnectionStateListener, DeviceStatusChangeListener {
+
+
+    protected Logger logger = LoggerFactory.getLogger(YeelightHandlerBase.class);
+    protected DeviceBase mDevice;
+
+    public YeelightHandlerBase(Thing thing) {
+        super(thing);
+    }
+
+    protected void updateUI(DeviceStatus status) {
+        thing.setLabel(status.getName().isEmpty() ? DeviceManager.getDefaultName(mDevice) : status.getName());
+    }
+
+    @Override
+    public void initialize() {
+        mDevice = DeviceFactory.build(getDeviceModel(getThing().getThingTypeUID()).name(), getThing().getUID().getId());
+        mDevice.setDeviceName(getThing().getLabel());
+        mDevice.setAutoConnect(true);
+        DeviceManager.getInstance().addDevice(mDevice);
+        mDevice.registerConnectStateListener(this);
+        mDevice.registerStatusChangedListener(this);
+        updateStatus(ThingStatus.OFFLINE);
+        DeviceManager.getInstance().startDiscovery(15 * 1000);
+    }
+
+    private DeviceType getDeviceModel(ThingTypeUID typeUID) {
+        if (typeUID.equals(YeelightBindingConstants.THING_TYPE_CEILING)) {
+            return DeviceType.ceiling;
+        } else if (typeUID.equals(YeelightBindingConstants.THING_TYPE_WONDER)) {
+            return DeviceType.color;
+        } else if (typeUID.equals(YeelightBindingConstants.THING_TYPE_DOLPHIN)) {
+            return DeviceType.mono;
+        } else if (typeUID.equals(YeelightBindingConstants.THING_TYPE_STRIPE)) {
+            return DeviceType.stripe;
+        } else {
+            return null;
+        }
+    }
+
+
+
+    @Override
+    public void onConnectionStateChanged(ConnectState connectState) {
+        logger.debug("onConnectionStateChanged -> {}", connectState.name());
+        switch (connectState) {
+            case DISCONNECTED:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Device is offline !");
+                if (mDevice.isAutoConnect()) {
+                    DeviceManager.sInstance.startDiscovery(5 * 1000);
+                }
+                break;
+
+            case CONNECTED:
+                updateStatus(ThingStatus.ONLINE);
+                mDevice.queryStatus();
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("ChannelLinked -> {}",channelUID.getId());
+        super.channelLinked(channelUID);
+
+        if (channelUID.getId().equals(YeelightBindingConstants.CHANNEL_BRIGHTNESS)) {
+
+            Runnable task = () -> {
```

Comment:
Please use the scheduler which is available in the parent class do not use the sleep but schedule with a delay instead.